### PR TITLE
Feature/automated panelite tests

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Going through the notebooks of Panelite to identify which notebooks which to `piplite.install` what. And which notebooks cannot run without errors has been a pain.

I would like to automate the process of testing the Panelite notebooks for errors. This is an attempt.

- [x] Open Panelite (either online or locally)
- [x] Navigate to notebook
- [x] Run all cells
- [x] Determine if an exception is raised
- [x ] Retry up to 3 times if "Run All" never finishes. But fail if an exception is raised.
- [x] Minimize flakyness of tests
- [ ] Maximize speed of tests
- [ ] Enable to test new build of Panelite (how do I build panelite in Panel?)
- [x] Functionality to loop through list of relevant notebooks.
- [ ] Make sure 100% of tests pass

Works like below (2x speed)

https://user-images.githubusercontent.com/42288570/232246575-11ee3423-804a-48b0-9833-34c6c2a9b891.mp4

### CODE REVIEW

**Comment by MarcSkovMadsen** [comment]
In order to be able to test the `main` branch or something close I need the panel and bokeh wheel files for Panelite.

Where do I find the ones I can use for testing the main branch or the latest beta or release candidate @philippjfr ?

image

**Comment by MarcSkovMadsen** [comment]
I've tried building Panelite/ Jupyterlite. But unfortunately it fails. I've reported the issue here https://github.com/jupyterlite/jupyterlite/issues/1055.

**Comment by MarcSkovMadsen** [comment]
Ok a fresh virtual environment solved my issue and I can build Panelite.

But when I serve it using `python -m http.server` it does not show any notebook files ??

image

File: scripts/generate_panelite_content.py [hunk 1]
```diff
@@ -42,13 +81,20 @@
     "reference/widgets/RangeSlider.ipynb": ["https://github.com/holoviz/panel/issues/4402"],
     "reference/widgets/SpeechToText.ipynb": ["https://github.com/holoviz/panel/issues/4404"],
     "reference/widgets/Terminal.ipynb": ["https://github.com/holoviz/panel/issues/4407"],
-    "gallery/components/VuePdbInput.ipynb": ["https://github.com/holoviz/panel/issues/4417"],
+    "user_guide/Async_and_Concurrency.ipynb": ["https://github.com/pyodide/pyodide/issues/237"],
+    "user_guide/Performance_and_Debugging.ipynb": ["https://github.com/aio-libs/aiohttp/issues/7253"],
```
Comment (philippjfr) [inline scripts/generate_panelite_content.py:85]: We don't have user guides anymore.

File: scripts/generate_panelite_content.py [hunk 2]
```diff
@@ -17,15 +17,54 @@
...
+    "gallery/simple/temperature_distribution.ipynb": ["https://github.com/holoviz/panel/issues/4393"],
+    "gallery/simple/xgboost_classifier.ipynb": ["https://github.com/holoviz/panel/issues/4416"],
+    "gallery/streaming/hardware_automation.ipynb": ["https://github.com/holoviz/panel/issues/4393"],
+    "gallery/streaming/hardware_automation.ipynb": ["https://github.com/holoviz/panel/issues/4393"],
+    "gallery/streaming/streaming_indicator.ipynb": ["https://github.com/holoviz/panel/issues/4416"],
+    "gallery/streaming/streaming_perspective.ipynb": ["https://github.com/holoviz/panel/issues/4416"],
+    "gallery/styles/PlotlyStyle.ipynb": ["https://github.com/holoviz/panel/issues/4416"],
+    "gallery/styles/SeabornStyle.ipynb": ["https://github.com/holoviz/panel/issues/4615"],
+    "gallery/viz/GraphViz.ipynb": ["https://github.com/xflr6/graphviz/issues/197"],
+    "gallery/viz/NetworkX.ipynb": ["https://github.com/pygraphviz/pygraphviz/issues/453"],
```
Comment (philippjfr) [inline scripts/generate_panelite_content.py:63]: All of these are being moved and/or removed. I'll update the PR when I'm done with that.

**Comment by philippjfr** [comment]
This is great btw but should probably be part of the regular UI test suite (or run as a separate job) rather than being a script.

**Comment by MarcSkovMadsen** [comment]
Lets get it working and then we can move it around.

But it take some time to run, so I 'm not sure you want to run it always.

**Comment by MarcSkovMadsen** [comment]
I'm now adding the dependencies for the new notebooks. I can see that you (@philippjfr) are very often reading `.parquet` files. This is great for data science. But Its not great for Panelite as you cannot read parquet files via pandas.read_parquet 😄 

So all those notebooks won't work.

See https://github.com/apache/arrow/issues/34996

**Comment by MarcSkovMadsen** [comment]
FYI. @philippjfr . I've updated the Panelite file generation script to list the notebooks without dependencies defined. 

image

This way the dictionary of notebooks/ dependencies will be easy to maintain.

**Comment by maximlt** [comment]
Seems like `fastparquet` is available on pyodide https://github.com/pyodide/pyodide/pull/3590

**Comment by MarcSkovMadsen** [comment]
`fastparquet` seems to work.

image

I can't seem to figure out if `pyodide-http` is needed. I read somewhere its now a part of Pyodide. I also think I've read somewhere its built into Panelite. But for the gallery/featured/nyc_deckgl.ipynb notebook I've had to import it specifically for it to work ????

**Comment by philippjfr** [comment]
Note that I've **completely** overhauled the gallery and may be re-adding specific examples I delete back to how_to examples. I'll also add a script to turn the how-to markdown files into notebooks. So there's a lot of churn that will have to happen here, sorry about that.

**Comment by MarcSkovMadsen** [comment]
I've added an info note to the Panelite notebooks. This improves the user experience significantly as you know what to expect.

image

image

**Comment by philippjfr** [comment]
I'm going to merge. I must admit though I haven't actually run your tests. Will work on that in another PR.

**Comment by MarcSkovMadsen** [comment]
To me it seems the newly released Panelite did not run on my changes @philippjfr ? There is no first info cell and the `installed` packages are not what I've been listing.

image

**Pull request merged**